tests/utils/local_network_setup.go

```diff
@@ -163,55 +169,110 @@ func SetupNetwork(warpGenesisFile string) {
 	SetupProposerVM(ctx, fundedKey, manager, 0)
 	SetupProposerVM(ctx, fundedKey, manager, 1)
 
-	// Set up subnet URIs
+	// Create the ANR client
+	logLevel, err := logging.ToLevel("info")
+	Expect(err).Should(BeNil())
+
+	logFactory := logging.NewFactory(logging.Config{
+		DisplayLevel: logLevel,
+		LogLevel:     logLevel,
+	})
+	zapLog, err := logFactory.Make("main")
+	Expect(err).Should(BeNil())
+
+	anrClient, err = runner_sdk.New(runner_sdk.Config{
+		Endpoint:    "0.0.0.0:12352",
+		DialTimeout: 10 * time.Second,
+	}, zapLog)
+
+	SetSubnetValues()
+
+	log.Info("Finished setting up e2e test subnet variables")
+}
+
+func SetSubnetValues() {
 	subnetIDs := manager.GetSubnets()
 	Expect(len(subnetIDs)).Should(Equal(2))
 
 	subnetA = subnetIDs[0]
-	subnetADetails, ok := manager.GetSubnet(subnetA)
-	Expect(ok).Should(BeTrue())
-	Expect(len(subnetADetails.ValidatorURIs)).Should(Equal(5))
-	blockchainIDA = subnetADetails.BlockchainID
-	chainANodeURIs = append(chainANodeURIs, subnetADetails.ValidatorURIs...)
+	SetSubnetAValues(subnetA)
 
 	subnetB = subnetIDs[1]
-	subnetBDetails, ok := manager.GetSubnet(subnetB)
+	SetSubnetBValues(subnetB)
+}
+
+func SetSubnetBValues(subnetID ids.ID) {
```

Comment:
Is it possible to use the same function to set the subnet values? `SetSubnetBValues` and `SetSubnetAValues` are similar.

Reply:
We can totally clean this all up into a loop. I don't mind making a PR to general-e2e after this is merged in.

Reply:
Agreed. Let's save general cleanup work for https://github.com/ava-labs/teleporter/pull/93

Reply:
Addressed in #93